.github/scripts/validate-e2e-page-object-usage.ts

```diff
@@ -0,0 +1,131 @@
+
+                await downloadArtifact(changedFilesArtifactUrl, outputDir);
+
+                changedFilesContent = readFileContent(outputDir);
+
+                if (changedFilesContent) {
+                    console.log('Artifact downloaded and read successfully.');
+                    break;
+                }
+            } catch (error) {
+                console.error(`Error fetching artifact: ${error.message}`);
+            }
+
+            attempts++;
+            if (attempts < maxAttempts) {
+                console.log(`Retrying in 15 seconds... (${attempts}/${maxAttempts})`);
+                await sleep(15);
+            }
+        }
+
+        if (!changedFilesContent) {
+            console.error('No artifacts found for changed files. Exiting with failure.');
+            process.exit(1);
+        }
+
+        // Parse the changed and new files with status
+        const changedAndNewFilePathsWithStatus = changedFilesContent.split('\n').filter(line => line.trim() !== '').map(line => {
+            const [status, filePath] = line.split('\t');
+            return { status, filePath };
+        });
+
+        // Filter files based on the provided fileStatus
+        const filteredFiles = changedAndNewFilePathsWithStatus.filter(file => {
+            if (fileStatus === 'both') {
+                return file.status === 'A' || file.status === 'M';
+            }
+            return file.status === fileStatus;
+        }).map(file => file.filePath);
+
+        e2eFiles = filterE2eChangedFiles(filteredFiles);
+        console.log('Filtered E2E files:', e2eFiles);
+    } else {
+        // Running locally
+        console.log('Running locally, performing git diff against main branch...');
+        const diffOutput = execSync('git diff --name-status main...HEAD').toString().trim();
+        const changedFiles = diffOutput.split('\n').filter(line => line.trim() !== '').map(line => {
+            const [status, filePath] = line.split('\t');
+            return { status, filePath };
+        });
+
+        // Filter files based on the provided fileStatus
+        const filteredFiles = changedFiles.filter(file => {
+            if (fileStatus === 'both') {
+                return file.status === 'A' || file.status === 'M';
+            }
+            return file.status === fileStatus;
+        }).map(file => file.filePath);
+
+        e2eFiles = filterE2eChangedFiles(filteredFiles);
+        console.log('Filtered E2E files:', e2eFiles);
+    }
+
+    if (e2eFiles.length === 0) {
+        console.log('No E2E files to validate. Exiting successfully.');
+        process.exit(0);
+    }
+
+    // Check each E2E file for page object usage
+    for (const file of e2eFiles) {
+        try {
+            const content = fs.readFileSync(file, 'utf8');
+            // Check for the presence of page object imports
+            const usesPageObjectModel = content.includes('./page-objects/');
```

Comment:
this has an underlying assumption that all specs that use POM will have at least 1 page-objects import, and that those who don't, won't have any.
This could be 'tricked' or not apply in some cases but it's hard to tell when would this happen

Reply:
agree, we can keep it like this for now and adjust it in the future if any exceptions arise